### ISSUE TITLE
CA2008 Do not create tasks without passing a TaskScheduler doesn't appear when pasing method group

### ISSUE THREAD

**Issue by NtFreX** [dotnet/roslyn-analyzers#1908]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

Example: v2.6.2

#### Diagnostic ID

`CA2008: Do not create tasks without passing a TaskScheduler`
> Do not create tasks unless you are using one of the overloads that takes a TaskScheduler. The default is to schedule on TaskScheduler.Current, which would lead to deadlocks. Either use TaskScheduler.Default to schedule on the thread pool, or explicitly pass TaskScheduler.Current to make your intentions clear.

I could not find a documentation page.

#### Repro steps

Use `TaskFactory.StartNew` as method group parameter.

```
public static class Program
{
    public static async Task Main(string[] args)
    {
        var taskScheduler = TaskScheduler.Default;
        var taskFactory = new TaskFactory(taskScheduler);
        var actions = new Action[] { Console.WriteLine, Console.WriteLine };
        await Task.WhenAll(actions.Select(taskFactory.StartNew)).ConfigureAwait(false);
    }
}
```

#### Expected behavior

Waring is displayed.

#### Actual behavior

No warning is displayed.

----------

*I just stumpled upon this "behaivor". It's not important for me I just wanted to report it for documentation reasons.*

**Comment by AArnott** [comment]
Amusingly, our version of this analyzer (VSTHRD105) got [the same feedback](https://github.com/Microsoft/vs-threading/issues/370).

**Comment by AArnott** [comment]
Oh wait, that's not the same issue. You're arguing that the diagnostic *should* appear with a method group. But per the issue I linked to, in this particular example case (of using a custom `TaskFactory`) it may be good to suppress such a diagnostic.

**Comment by NtFreX** [comment]
@AArnott Hey I think this one is different but it could be the same as this one.
[CA2008 Do not create tasks without passing a TaskScheduler should not appear](https://github.com/dotnet/roslyn-analyzers/issues/1907)
